plugin/driver/network_utils.go

```diff
@@ -134,3 +134,123 @@ func getGratuitousArp(mac net.HardwareAddr, ip net.IP) []byte {
 
 	return buf.Bytes()
 }
+
+type NetworkIfc struct {
+	ifc_name string
+	mac_addr string
+	ip       string
+	mask     string
+}
+
+func GetInterfaceInformation(net_ns_path string) ([]NetworkIfc, error) {
+
+	var netns ns.NetNS
+	var err error
+
+	netns, err = ns.GetNS(net_ns_path)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open netns %s: %v", net_ns_path, err)
+	}
+	defer netns.Close()
+
+	var net_info []NetworkIfc
+
+	// Let's go in
+	err = netns.Do(func(_ ns.NetNS) error {
+		// Get Ifcs on container
+		ifc_list, err := net.Interfaces()
+
+		if err != nil {
+			return fmt.Errorf("Failed to get interfaces in Netns: %v", err)
+		}
+
+		// For some reason I can't properly copy the full interface info struct outside. Seems the go net library is doing some
+		// caching inbetween (checked on source code)
+
+		// This is pathetic, but until I have a way to copy the full struct and substructs, I have to do this.
+		// Quite the chance the return is not static data, but instead is computed every time yout call, hence data changes between
+		// invocations in and out of the netNS
+		for _, ifc := range ifc_list {
```

Comment:
I somehow agree but, you will mess up the context of each one and end up with a huge function doing several stuff, which is unadviced. Right now it clearly separates the two routines: one does one job, the other one the other.

Additionally, even though the information used is more or less the same, the operations done on arping are inside the namespace, whilst the metadata write has to be done outside. This is only my way of seeing it, if you guys still think to merge both, we can do.